Core/SubscriptionManager/src/test/java/eu/neclab/ngsildbroker/subscriptionmanager/service/SubscriptionServiceTest.java

```diff
@@ -1,146 +1,280 @@
+		subscription.setNotification(notifyParam);
+		multimaparr.put("content-type", "application/json");
+		ArrayList<String> result = new ArrayList<String>();
+
+		SubscriptionRequest subRequest = new SubscriptionRequest(subscription, context, multimaparr, 0);
+		tenant2subscriptionId2Subscription.put(subRequest.getTenant(), subRequest.getId().toString(), subRequest);
+		ReflectionTestUtils.setField(baseSubscriptionService, "tenant2subscriptionId2Subscription",
+				tenant2subscriptionId2Subscription);
+
+		// PowerMockito.spy(BaseSubscriptionInfoDAO.class);
+		// PowerMockito.doReturn(result).when(subscriptionInfoDAO.getStoredSubscriptions());
+
+		// when(subscriptionInfoDAO.getStoredSubscriptions()).thenReturn(result);
+		// Method postConstruct =
+		// BaseSubscriptionService.class.getDeclaredMethod("loadStoredSubscriptions");
+		// postConstruct.setAccessible(true);
+		// postConstruct.invoke(baseSubscriptionService);
+		// multimaparr.put("content-type", "application/json");
+		// manager.unsubscribe("urn:ngsi-ld:Subscription:173223", multimaparr);
+		// verify(tenant2subscriptionId2Subscription, times(1)).get(any(), any());
+		// verify(manager, times(1)).subscribe(any());
+	}
+
+	/**
+	 * this method is used to test getSubscription method of SubscriptionService
+	 * class
+	 * 
+	 * @throws URISyntaxException
+	 */
+	@Test
+	public void getSubscriptionTest() throws URISyntaxException {
+		String errorMessage = null;
+		Subscription removedSub = new Subscription();
+		// Table<String, String, SubscriptionRequest> tenant2subscriptionId2Subscription
+		// = HashBasedTable.create();
+		URI id = new URI("urn:ngsi-ld:Subscription:173223");
+		// when(subscriptionId2Subscription.remove(any())).thenReturn(removedSub);
+		List<Object> context = new ArrayList<>();
+		NotificationParam notifyParam = new NotificationParam();
+		EndPoint endPoint = new EndPoint();
+		URI uri = new URI("http://localhost:8080");
+		endPoint.setUri(uri);
+		endPoint.setAccept("application/json");
+		notifyParam.setEndPoint(endPoint);
+		Format format = Format.normalized;
+		format = Format.keyValues;
+		notifyParam.setFormat(format);
+		Subscription subscription = new Subscription();
+		subscription.setId("urn:ngsi-ld:Subscription:173223");
+		subscription.setType("Subscription");
+		subscription.setNotification(notifyParam);
+		multimaparr.put("content-type", "application/json");
+		SubscriptionRequest subRequest = new SubscriptionRequest(subscription, context, multimaparr, 0);
+		tenant2subscriptionId2Subscription.put(subRequest.getTenant(), subRequest.getId().toString(), subRequest);
+		ReflectionTestUtils.setField(baseSubscriptionService, "tenant2subscriptionId2Subscription",
+				tenant2subscriptionId2Subscription);
+
+		try {
+			manager.getSubscription("urn:ngsi-ld:Subscription:173223", multimaparr);
+		} catch (ResponseException e) {
+			errorMessage = e.getMessage();
+		}
+		Assert.assertEquals(errorMessage, "urn:ngsi-ld:Subscription:173223 not found");
```

Comment:
[Major] why you compare with "not found" but I think it should be return id.

Reply:
it throws response exception so we compare inside catch block